### ISSUE TITLE
Allow addons to be signed by NVAccess

### ISSUE THREAD

**Issue by dawidpieper** [nvaccess/nvda#11603]
### Is your feature request related to a problem? Please describe.
There are many doubts related to NVDA Addons security, observed at universities or in companies.
We are prompted to download addons only from known sources, but it may be too less, depending on organization security policies.

### Describe the solution you'd like
I think that NVAccess should invistigate procedure of signing NVDA addons. Users obviously should still be able to download addons that are not signed, but it would be nice to see an additional warning or to provide checkbox in settings enabling only installation of the addons trusted by NVAccess.
Sometimes links at addons repository point to external domains, for example in case of NVDA Remote. It is unlikely that somebody hacks such a domain, but it cannot be neglected.

### Describe alternatives you've considered
Obviously, addon provider can include checksums, but it would not be so clear for average users.
The solution would be to **always** download addons only from official repository, but it is not always possible.

### Additional context
NVDA addons are zip archives, so I can see two basic options:
1. to create a new format (something like nvda-addon-signed), that would include zip archive and it's signed sha2 checksum,
2. to include file like sign.csr that would include shortcuts of all addon files. This could be also used to check addon integrity after NVDA starts.

**Comment by feerrenrut** [comment]
Thanks for the feature request @dawidpieper 

I don't think this is likely in the short term. It's important that users understand the security risks associated with installing add-ons. It should be noted that NV Access does not currently inspect 3rd party add-ons, and can not guarantee the safety of using such add-ons. We have plans to develop the add-ons "submission" process and infrastructure, see https://github.com/nvaccess/addon-store-submission.

I don't think this issue belongs on the NVDA issue tracker, I'll close it for now.
